pkg/metrics/metrics.go

```diff
@@ -0,0 +1,65 @@
+package metrics
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var counterVecs = make(map[MetricType]*prometheus.CounterVec)
```

Comment:
Rather than a global, it would be more idiomatic Go to have a `NewCounterVecMap` function that returns a (probably opaque) *struct that contains the allocation.

That pointer would be stored in Controller. And then InitAllCounterVecs and IncrementCounterVec would be methods on *CounterVecMap.

Otherwise we have a singleton global that can be harder to test and assumes that there is only ever 1 such map in an image.